### PR TITLE
DCOS-18993: implement new select placement

### PR DESCRIPTION
**Pull request by juliangieseke**
---

~⚠️ Depends on https://github.com/dcos/dcos-ui/pull/2577.~ merged 👍 

---

~⚠️ Since #2531 has copied tests which are affeced by this change, this PR is on hold until #2531 is merged.~ #2531 is merged

---

~I have started to implement the new `<Select>`, unfortunately I wasn't able to find the needed CSS code, maybe someone else can pair up with me tomorrow (or fix it, if it is done in 5min)~ 
~Currently the button is grey (as in `reactjs-components`), but it has to be white and fixed in width~

Button looks as expected now and the dropdown is implemented. If someone knows how to do the css part in a better way - please tell me :) 

--- 

To test this / help with:
1. checkout this branch
2. `npm start`
3. disable placement plugin
4. navigate to services -> create -> placement -> add advanced constraint

### CODE REVIEW

**Comment by orlandohohmeier** [comment]
@juliangieseke #2527 was merged.

**Comment by juliangieseke** [comment]
Adding hold merge label because PR #2531 has copied test cases, I need to change them after it landed to prevent failing tests.

**Comment by juliangieseke** [comment]
done 👍

**Comment by nLight** [comment]
I still see discrepancy with designs.

1. Normal vs bold as a selected label
2. When you hover an item margins for the selection is inconsistent, more space at the bottom, should be equal
3. Color of the description is off

You can also look at this css mockup https://jsfiddle.net/wkx2kn4g/ provided by @leemunroe 

Design
screen shot 2017-12-04 at 10 57 26

Implementation:
screen shot 2017-12-04 at 10 54 12

**Comment by juliangieseke** [comment]
@ahoskins I'm not able to reproduct your error, did you `npm install` lately (to get the newest version of `reactjs-components`)?

**Comment by juliangieseke** [comment]
design is adapted 👍 
tests from #2531 are also adapted 🤞

**Comment by juliangieseke** [comment]
Alfred test this please

**Comment by juliangieseke** [comment]
re 1: hm, thats weird, isnt the formpartial re-rendered if something changes? 
re 2: thats basicly a feature request against our Dropdown - I dont think it is implemented yet 🤔

**Comment by ahoskins** [comment]
I believe the issue is the `Select` is not updating it's state when the value from props changes. See in below screenshot (after editing JSON):
screen shot 2017-12-06 at 11 37 48 am

I think we need to add `componentWillReceiveProps` to Select, and update it's `this.state.value` when it's `this.props.value` changes.

**Comment by nLight** [comment]
Confirmed @ahoskins. We have to fix `reactjs-components` @juliangieseke

**Comment by nLight** [comment]
@juliangieseke re1: re-renders in react way. Meaning that if a component doesn't have to be updated it won't be updated. Since we get the initial value from the props and don't listen to the props change - we never update the component.

**Comment by ahoskins** [comment]
@bstavroulakis and I rebased this and tried to test this with the fixed Select component, and getting this error:

screen shot 2017-12-14 at 5 13 01 pm

**Comment by juliangieseke** [comment]
updated implementation to the latest reactjs-components update - it is ready to go once https://github.com/dcos/dcos-ui/pull/2577 merges 👍

File: plugins/services/src/js/constants/OperatorTypes.js [hunk 1]
```diff
@@ -2,37 +2,51 @@ const OperatorTypes = {
...
+    stringNumberValue: true,
+    name: "Group By",
+    description: "Run app tasks evenly distributed across a certain attribute"
   },
   IS: {
     requiresValue: true,
     requiresEmptyValue: false,
-    stringNumberValue: false
+    stringNumberValue: false,
+    name: "Is",
```
Comment (ahoskins) [inline plugins/services/src/js/constants/OperatorTypes.js]: 🤔  Is this suppose to be the same description as "cluster"?

Comment (juliangieseke) [reply plugins/services/src/js/constants/OperatorTypes.js]: good catch! probably not, but thats all we have (see screenshot by dima above), can you maybe check back with lily for a new text?

**Comment by ahoskins** [comment]
I sat with @lilysamimi and showed this. We made minor tweaks to the description (capitalize ID, and write "number" instead of #). And changed the selected option to highlight the title in purple. This aligns more with cnvs styleguide.

Current behavior: https://cl.ly/2g2Y0h101K1s

File: plugins/services/src/js/constants/OperatorTypes.js (hunk 1 above)
Comment (ahoskins) [reply plugins/services/src/js/constants/OperatorTypes.js]: We will have to follow-up with someone from docs on this. The other descriptions came from here: http://mesosphere.github.io/marathon/docs/constraints.html, but IS is not on that page so we don't have a good idea what the description is.

Comment (ahoskins) [reply plugins/services/src/js/constants/OperatorTypes.js]: I suggest not blocking this PR, and make a small follow-up ticket to improve this description.

**Comment by juliangieseke** [comment]
Alfred test this

File: plugins/services/src/js/constants/OperatorTypes.js (hunk 1 above)
Comment (orlandohohmeier) [reply plugins/services/src/js/constants/OperatorTypes.js]: @juliangieseke @ahoskins did one of you file an issue?

**Comment by orlandohohmeier** [comment]
LGTM 💯 ✨

**Pull request merged**